datamade/bga-payroll
# Update data making and a few odds and ends on the site to display 2018 data

## Description

This PR:

- Adds a step to the pre-processing pipeline to strip leading whitespace from incoming data. This is a stopgap measure that should be more permanently addressed in #384.
- Updates the date validation script to:
    - Strip whitespace when checking date format.
    - Stop correcting future dates, as it's a violation of our don't alter source data principle. Instead, the script suppresses future dates.
- Updates the FOIA source lookup to cover all responding agencies in the 2018 data.
- Only selects not null salaries to render the chart on the homepage.
- Makes the data year temporarily configurable. This is also a stopgap measure that should be more permanently addressed when multi-year support is added. 

Closes #372.

## Testing instructions

- Run the app locally, as instructed in the README.
- If your local install contains 2018 data, add `DATA_YEAR = 2018` to your local settings file. If this is not configured, `DATA_YEAR` will default to 2017.
- Check the homepage and a couple of unit and department pages and confirm that you don't see any (unexpected) errors.
